src/dune_engine/build_system.ml

```diff
@@ -1483,7 +1477,7 @@ end = struct
       Fiber.parallel_iter_set
         (module Path.Set)
         chdirs
-        ~f:(fun p -> Memo.Build.run (Fs.mkdir_p_or_check_exists ~loc p))
+        ~f:(fun p -> Memo.Build.run (Fs.mkdir_p_or_assert_existence ~loc p))
```

Comment:
Hmm, I thought it's a bit weird that we "assert_existence" of the same non-build directories twice: here and 20 lines earlier. I guess it's fine.

Reply:
Indeed. I refactored the code to avoid duplicate work.